unify symbolic_shapes and sizevars dynamic shapes APIs naming 1

laithsakka opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #155776
* __->__ #154774

Inductor have a set of APIs that allows performing symbolic evaluations similar to that of symbolic shapes 
but it operates on sympy expressions instead of symnodes. Namings are not consistent making them consistent 
in this stack.

Step 1 : unify statically_know_true naming! for consistent experience. 

cc @ezyang @SherlockNoMad @EikanWang @jgong5 @wenzhe-nrv @voznesenskym @penguinwu @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

laithsakka committed:
unify symbplicshapes and sizevars APIs namings 1

[ghstack-poisoned]

laithsakka committed:
Update on "unify symbplic_shapes and sizevars APIs dynamic shapes APIs namings 1"




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

laithsakka committed:
Update on "unify symbplic_shapes and sizevars APIs dynamic shapes APIs namings 1"




cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

drisspg commented on torch/_inductor/sizevars.py:
```diff
@@ -307,44 +303,21 @@ def prune(index):
+    # asked questions can be answered without guarding otherwise they return False.
+    # Those are similar to statically_known_true in symbolic_shapes but operate on sympy expressions 
+    # instead of symnodes.
+    def statically_known_true(self, expr: Union[sympy.Basic, bool]) -> bool:
```
could you also add this in a doc block  to this function, also can you keep the `Note`

laithsakka replied:
the old note? why would someone be interested in how things were long ago?

drisspg commented on torch/_inductor/sizevars.py:
```diff
@@ -307,44 +303,21 @@ def prune(index):
         self, left: Union[Expr, int], right: Union[Expr, int]
     ) -> bool:
         """
         Returns a bool indicating if it is sound to optimize as if left and right are equal.
```
I kinda which all of these said something in a doc block , like `This function does not add guards, if it can determine that the expr left equals right at compile time it will return True, otherwise it will return False. An example when you should use is: a performance optimization that won't hinder correctness, etc...'

drisspg reviewed (approved):
Left some comments

laithsakka committed:
Update on "unify symbolic_shapes and sizevars dynamic shapes APIs naming 1"


Inductor have a set of APIs that allows performing symbolic evaluations similar to that of symbolic shapes 
but it operates on sympy expressions instead of symnodes. Namings are not consistent making them consistent 
in this stack.

Step 1 : unify statically_know_true naming! for consistent experience. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

bobrenjc93 reviewed (approved):
thanks for doing this

eellison reviewed (approved):
nice !

laithsakka committed:
Update on "unify symbolic_shapes and sizevars dynamic shapes APIs naming 1"


Inductor have a set of APIs that allows performing symbolic evaluations similar to that of symbolic shapes 
but it operates on sympy expressions instead of symnodes. Namings are not consistent making them consistent 
in this stack.

Step 1 : unify statically_know_true naming! for consistent experience. 

cc ezyang SherlockNoMad EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge
